### PR TITLE
NOOS-415/upgrade-python

### PR DESCRIPTION
**Pull request by zyassine**
# Description

**JIRA Reference**: [NOOS-415](https://noosenergy.atlassian.net/browse/NOOS-415)

[NOOS-415]: https://noosenergy.atlassian.net/browse/NOOS-415?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docker/jupyterlab/environment.yml [hunk 1]
```diff
@@ -27,3 +27,4 @@ dependencies:
   - pip
   - pip:
     - jupyterlab-spreadsheet-editor
+  - matplotlib
```
Comment (pieeri) [inline docker/jupyterlab/environment.yml:28]: 🐼 

To regroup under the "plotting" dependency category.

File: docker/circleci/Dockerfile [hunk 2]
```diff
@@ -30,7 +30,7 @@ RUN npm install --global yarn@${YARN_VERSION}
 
 # Install Python package managers
 # -------------------------------
-ARG PIPENV_RELEASE="2023.9.8"
+ARG PIPENV_RELEASE="2023.6.2"
```
Comment (pieeri) [inline docker/circleci/Dockerfile:33]: 🐼 

Would it be cleaner to squash such a commit into the previous one?

File: docker/jupyterlab/environment.yml [hunk 3]
```diff
@@ -12,12 +12,10 @@ dependencies:
   - xeus-python
   # plotting
   - plotly
-  # https://github.com/plotly/jupyter-dash/issues/75
-  - dash<2.0.0
+  - dash<2.10
```
Comment (pieeri) [inline docker/jupyterlab/environment.yml:15]: ❓ 

Is it required to include such a change with the upgrade towards Python `3.11.*` or could we ship Dash `2.*` into a subsequent PR (in conjonction with changes in `dash-neptune-mdc`)?

File: docker/circleci/Dockerfile [hunk 4]
```diff
@@ -1,12 +1,12 @@
 # https://circleci.com/developer/images/image/cimg/python
 # https://github.com/CircleCI-Public/cimg-python
 # https://hub.docker.com/r/cimg/python
-FROM cimg/python:3.8.17
+FROM cimg/python:3.11.4
 
 USER root
 
 # # Provision Selenium (Firefox)
```
Comment (pieeri) [inline docker/circleci/Dockerfile:8]: 🐼 

Non-compliant comment formatting.

**Review by pieeri — CHANGES_REQUESTED** [review]
Quick question in relation to Dash version upgrade.

File: docker/jupyterlab/environment.yml (hunk 3 above)
Comment (zyassine) [reply docker/jupyterlab/environment.yml:15]: I had an issue with pyyaml and pipenv. The solution I found is to upgrade pyyaml and I can't do it without Dash upgrade

**Review by pieeri — COMMENTED** [review]
LGTM

**Review by pieeri — APPROVED** [review]
Let's start rolling Python `3.11.*`!

**Pull request merged**